server/src/domain/media/media.service.ts

```diff
@@ -330,7 +378,20 @@ export class MediaService {
   }
 
   ensureThumbnailPath(asset: AssetEntity, extension: string): string {
-    const folderPath = this.storageCore.getFolderLocation(StorageFolder.THUMBNAILS, asset.ownerId);
+    return this.ensurePath(asset, StorageFolder.THUMBNAILS, extension);
+  }
+
+  ensureEncodedVideoPath(asset: AssetEntity, extension: string): string {
+    return this.ensurePath(asset, StorageFolder.ENCODED_VIDEO, extension);
+  }
+
+  private ensurePath(
+    asset: AssetEntity,
+    folder: StorageFolder.ENCODED_VIDEO | StorageFolder.UPLOAD | StorageFolder.PROFILE | StorageFolder.THUMBNAILS,
+    extension: string,
+  ): string {
+    let folderPath = this.storageCore.getFolderLocation(folder, asset.ownerId);
+    folderPath = join(folderPath, asset.id.substring(0, 2), asset.id.substring(2, 4), asset.id.substring(4, 6));
```

Comment:
Do we need this much nesting? Most other times I've seen a system like this, it just nested one level with `substring(0, 2)`.

Reply:
I thought it wouldn't hurt and I don't know how many assets some people might have in their library ;)
The original idea in the issue was a depth of 2 and I don't see any disadvantage in a little deeper nesting?

Reply:
Since it's the thumbs folder this isn't particularly important, but if someone wanted to poke around these files deep nesting might make that a bit of a pain. I don't really have any strong points for/against though :)

Reply:
Tbh I can't really estimate how much a performance impact large directories actually make and how many assets would end up in one folder if nested 1, 2 or 3 deep. xD
Let's just wait for others and see what they think about :)

Reply:
Fixing all the tests with a nesting of 3 is a pain though haha
`upload/encoded-video/user-id/as/se/t-/asset-id.mp4` nice path lol

Reply:
I wrote a script to test with 150k random UUIDs.

depth=1 
585.9375 assets per leaf folder

depth=2
2.5508 assets per leaf folder

depth=3
1.0045 assets per leaf folder

So anything more than 1 layer deep puts almost every asset in its own folder.

Reply:
Which makes sense since the characters are hexadecimal so each two letter sequence has 16*16=256 combinations, so the number of assets in the leaf folders will be roughly `(num_assets) / 256^depth`.

Reply:
That makes sense. So even with ~1M assets there would still be like <15 (15 * 2^16 = something greater than 1M) assets per folder. Alright, so I guess it is perfectly fine if I just go with a depth of 2 instead.
Thanks @mertalev for testing!